app/models/salt_event_handlers/minion_start.rb

```diff
@@ -0,0 +1,13 @@
+class SaltEventHandlers::MinionStart
+  attr_reader :salt_event, :event_parsed_data
+
+  def initialize(salt_event)
+    @salt_event = salt_event
+    @event_parsed_data = @salt_event.parsed_data
+  end
+
+  def process_event
+    # false if a minion with this hostname already exists (uniqueness validation)
+    Minion.new(hostname: salt_event.parsed_data["id"]).save
```

Comment:
The difference is that ```create``` will always return the Minion while ```save``` will return ```false``` if no Minion was created so we know if a minion was created or not. On the other hand none of these is a proper return value for a method named ```process_event```. The event has been processed in both cases so the return value should probably be ```true``` but that doesn't provide us any usable information.